engineyard/ey-cookbooks-stable-v5
# [CC-1089] HTTPS param not set for FastCGI applications

## Description of your patch

Setting FastCGI HTTPS flag based on X-Forwarded-Proto header information

## Recommended Release Notes

None.

## Estimated risk

Low

## Components involved

Nginx

## Description of testing done

See QA instructions

## QA Instructions

1. Create simple PHP application. Simple How-To app (git://github.com/engineyard/howto.git) can be used for this testing.
2. Boot the environment with the latest V5 stack version. Environment should be run with following parameters:
  * Environment type - Single Instance.
  * SSL - enabled.
3. SSH to the instance.
4. Replace nginx cookbook with cookbook provided for this pull request. 
5. Run Chef `PATH=/usr/local/ey_resin/bin:$PATH /home/ey/bin/chef-solo -j /etc/chef/dna.json -c /etc/chef/solo.rb`
6. Run command `curl 127.0.0.1/phpinfo.php | grep HTTPS`
7. Ensure HTTPS option has no value.
8. Run command `curl  https://127.0.0.1/phpinfo.php | grep HTTPS`
9. Ensure HTTPS option value changed to 'on'.
10. Run command `curl -H"X-Forwarded-Proto: https" http://127.0.0.1/phpinfo.php | grep HTTPS`
